dotnet/src/SemanticKernel.Core/Planning/PlanExtensions.cs

```diff
@@ -68,13 +69,24 @@ public static string ToPlanString(this Plan plan, string indent = " ")
         return planString;
     }
 
+    /// <summary>
+    /// Returns decorated instance of <see cref="ISKFunction"/> with plan enabled instrumentation.
+    /// </summary>
+    /// <param name="plan">Instance of <see cref="Plan"/> to decorate.</param>
+    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    public static ISKFunction WithInstrumentation(this ISKFunction plan, ILoggerFactory? loggerFactory = null)
```

Comment:
While we want to work with `ISKFunction` in `InstrumentedPlan`, this extension method should still extend `Plan`, because we need some differentiation between Function and Plan, and there is already an extension method for Function:
https://github.com/microsoft/semantic-kernel/blob/0c1b669d234767c9a3b4839c8859e2ec548780cb/dotnet/src/SemanticKernel.Core/Extensions/SKFunctionExtensions.cs#L84-L87
```suggestion
    public static ISKFunction WithInstrumentation(this Plan plan, ILoggerFactory? loggerFactory = null)
```

Reply:
Reverted